in_toto/runlib.py

```diff
@@ -571,31 +574,67 @@ def in_toto_run(name, material_list, product_list, link_cmd_args,
       materials=materials_dict, products=products_dict, command=link_cmd_args,
       byproducts=byproducts, environment=environment)
 
-  link_metadata = Metablock(signed=link, compact_json=compact_json)
+  if use_dsse:
+    link_metadata = link.create_envelope()
 
-  signature = None
-  if signing_key:
-    LOG.info("Signing link metadata using passed key...")
-    signature = link_metadata.sign(signing_key)
+    signer = None
+    if signing_key:
+      LOG.info("Signing link metadata using passed key...")
+      signer = SSlibSigner(signing_key)
 
-  elif gpg_keyid:
-    LOG.info("Signing link metadata using passed GPG keyid...")
-    signature = link_metadata.sign_gpg(gpg_keyid, gpg_home=gpg_home)
+    elif gpg_keyid:
+      LOG.info("Signing link metadata using passed GPG keyid...")
+      signer = GPGSigner(keyid=gpg_keyid, homedir=gpg_home)
 
-  elif gpg_use_default:
-    LOG.info("Signing link metadata using default GPG key ...")
-    signature = link_metadata.sign_gpg(gpg_keyid=None, gpg_home=gpg_home)
+    elif gpg_use_default:
+      LOG.info("Signing link metadata using default GPG key ...")
+      signer = GPGSigner(keyid=None, homedir=gpg_home)
 
-  # We need the signature's keyid to write the link to keyid infix'ed filename
-  if signature:
-    signing_keyid = signature["keyid"]
-    filename = FILENAME_FORMAT.format(step_name=name, keyid=signing_keyid)
+    if signer:
+      signature = link_metadata.sign(signer)
+
+      signing_keyid = signature.keyid
+      filename = FILENAME_FORMAT.format(step_name=name, keyid=signing_keyid)
+
+      if metadata_directory is not None:
+        filename = os.path.join(metadata_directory, filename)
+
+      LOG.info("Storing link metadata to '{}'...".format(filename))
+      metadata_dict = link_metadata.to_dict()
+
+      indent = None if compact_json else 1
+      separators = (',', ':') if compact_json else (',', ': ')
+
+      with open(filename, 'w', encoding='utf8') as fp:
```

Comment:
Can this be offloaded to a `dump` method in the DSSE envelope?

Reply:
This is possible but Metadata wrapper for python-tuf uses different naming convention, `to_file` and `from_file`. In my opinion either this should be done in the in-toto itself or create a method that is suitable for both projects.

Reply:
I don't have a strong opinion vis-à-vis the naming convention, especially since the eventual plan for `Metablock` is to deprecate it. I think it makes sense for the DSSE library to implement it rather than the implementation extending it.